src/imu/imu.cpp

```diff
@@ -22,89 +25,109 @@ bool Imu::setup(yarp::os::Property& property)
     controlBoardOptions.put("localPortPrefix", "/test");
     controlBoardOptions.put("axesNames", axesNames.get(0));
+    
     ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(controlBoardDriver.open(controlBoardOptions), "Unable to open the controlBoard driver");
-
-    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(MASclientDriver.isValid(), "Device driver cannot be opened");
-    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(MASclientDriver.view(iorientation), "Unable to open orientation interface");
     ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(controlBoardDriver.isValid(), "Device driver cannot be opened");
     ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(controlBoardDriver.view(ipos), "Unable to open position control interface");
     ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(controlBoardDriver.view(ienc), "Unable to open encoder interface");
     ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(controlBoardDriver.view(iaxes), "Unable to open axes interface");
-    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(controlBoardDriver.view(ilim), "Unable to open limits interface");
-
-    outputPort.open("/test-imu/out");
-    ROBOTTESTINGFRAMEWORK_TEST_REPORT("Opening port "+outputPort.getName()+"...");
 
     ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(ienc->getAxes(&axes), "Cannot get number of controlled axes");
     std::string axisName;
     
-    for (int i = 0; i < axes; i++)
+    for (int axIndex = 0; axIndex < axes; axIndex++)
     {
-        ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(iaxes->getAxisName(i, axisName), "Cannot get the name of controlled axes");
+        ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(iaxes->getAxisName(axIndex, axisName), "Cannot get the name of controlled axes");
         axesVec.push_back(axisName);
     }
     
     ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(model.loadReducedModelFromFile(modelAbsolutePath.c_str(), axesVec), Asserter::format("Cannot load model from %s", modelAbsolutePath.c_str()));
     kinDynComp.loadRobotModel(model.model());
 
+    yarp::os::Property MASclientOptions;
+    MASclientOptions.put("device", "multipleanalogsensorsclient");
+    MASclientOptions.put("remote", portName);
+    MASclientOptions.put("local", "/imuTest"+portName);
+
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(MASclientDriver.open(MASclientOptions), "Unable to open the MAS client driver");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(MASclientDriver.isValid(), "Device driver cannot be opened");
+
+    yarp::os::Bottle & sensorsNamesList = sensorsList.addList();
+    yarp::os::Bottle *inputSensorsList;
+    inputSensorsList = property.find("sensorsList").asList();
+
+    if(inputSensorsList->size() == 0 || inputSensorsList->get(0).asString() == "all")
+    {
+        ROBOTTESTINGFRAMEWORK_TEST_REPORT("Testing all the IMUs available...");
+        sensorsNamesList.addString("l_arm_ft");
+        sensorsNamesList.addString("r_arm_ft");
+        sensorsNamesList.addString("head_imu_0");
```

Comment:
Why `all` add only those sensors? Maybe it is better to ask idyntree the list of all sensors, because those names can change

Reply:
> If I ask iDynTree for the list of the sensors of type `iDynTree::ACCELEROMETER`, it returns 62 sensors because it includes also the mtb
> 
> cc @traversaro

It would be great if we could test all the mtb4 sensors! Do you think there is anything preventing us to do so? I guess for known issues we could have an additional parameter to "exclude" sensors instead of adding them.

Reply:
Since so far the configuration files to read from the mtb are not updated (see for example https://github.com/robotology/icub-models/blob/devel/iCub/conf/MTB/gazebo_icub_left_arm_mtb.ini) and we should add all the 62 sensors to the `alljoints-inertials`, we could do a workaround: if we want to test `all` the sensors _available_, we could use another `IOrientationSensors` and make the `multipleanalogsensorsclient` do the view on this, so that we could retrieve only the orientation sensors that are listed in the `alljoints-inertials.xml` and exclude the mtb for the time being 

cc @Nicogene @traversaro